tools/api-markdown-documenter/src/documentation-domain-to-html/TransformationContext.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ConfigurationBase } from "../ConfigurationBase.js";
+import { defaultConsoleLogger } from "../Logging.js";
+import {
+	defaultTransformations,
+	type TransformationConfig,
+	type Transformations,
+} from "./configuration/index.js";
+
+/**
+ * Context passed down during recursive {@link DocumentationNode} rendering.
+ *
+ * @alpha
+ */
+export interface TransformationContext extends ConfigurationBase {
+	/**
+	 * Contextual heading level.
+	 *
+	 * @remarks
+	 *
+	 * Will automatically increment based on {@link SectionNode}s encountered, such that heading
+	 * levels can be increased automatically based on content hierarchy.
+	 */
+	headingLevel: number;
```

Comment:
Nit, seems like this could be readonly? I didn't see any direct assignments to a property of this name.

Reply:
That is a habit I need to get back into. I'm not sure when that stopped being my default mode of operation.